### PR TITLE
feat(ext/timers): add refTimer, unrefTimer API (alt)

### LINKED ISSUE #6141: Feature request:  Support async unref for timers
**Use case**:  As a a module developer, I want to run periodic tasks during the execution of my module.  This can be accomplished via `setTimeout` or `setInterval` for example.  However, I do not want these timers to block the exiting of my module.  `setInterval` in particular will cause the module to run forever until the `id` associated with it is cleared.

**Specific use case**:  `std/log` file handler now uses a buffer to write log messages.  Google and Go's glog implementations both use a log daemon to flush the buffer automatically every 30 seconds.  It would be good for `std/log` file handler to use `setInterval` to flush the buffer every 30 seconds as well to mimic this but not block module exit. (see also ref #6127)

For starters, there is already support for [async unref ops](https://github.com/denoland/deno/blob/master/core/ops.rs#L20) within Deno which would likely be used in this solution.

### PR DESCRIPTION
**Pull request by kt3k**
depends on #12862 (alternative to #12942)

This PR adds Deno.refTimer and Deno.unrefTimer APIs in the unstable namespace.

---
closes #6141

### CODE REVIEW

File: ext/timers/01_timers.js [hunk 1]
```diff
@@ -111,19 +112,20 @@
...
     } else {
       // TODO(@andreubotella): Deal with overflow.
       // https://github.com/whatwg/html/issues/7358
       id = nextId++;
-      cancelRid = core.opSync("op_timer_handle");
+      const cancelRid = core.opSync("op_timer_handle");
 
       // Step 4 in "run steps after a timeout".
-      MapPrototypeSet(activeTimers, id, cancelRid);
+      timerInfo = { cancelRid, isRef: true };
```
Comment (bnoordhuis) [inline ext/timers/01_timers.js:127]:
```diff
-       timerInfo = { cancelRid, isRef: true };
+       timerInfo = { cancelRid, isRef: true, promiseId: -1 };
```
Keeps the object monomorphic.

**Review by bnoordhuis — APPROVED** [review]
Needs a rebase but #12862 was merged earlier today and this PR otherwise LGTM.

File: ext/timers/01_timers.js [hunk 2]
```diff
@@ -255,6 +263,24 @@
     clearTimeout(id);
   }
 
+  function refTimer(id = 0) {
+    const timerInfo = MapPrototypeGet(activeTimers, id);
+    if (timerInfo.isRef) {
```
[the remark this answers is not in the document]
Comment (kt3k) [reply ext/timers/01_timers.js:268]: > if (timerInfo === undefined || timerInfo.isRef) { (ditto on line 277)

Oops. Thanks!

> The id = 0 default argument is kind of weird/footgun-y, I'd remove that.

I blindly followed the pattern used in clearTimeout/clearInterval, but I agree with your point

**Review by bartlomieju — APPROVED** [review]
LGTM too! Nice work @kt3k

**Comment by jcs224** [comment]
Is there any timeline this might go from unstable to stable? I think this might be the only thing preventing me from being able to push jobs from Deno Deploy to BullMQ

```
error: TypeError: Deno.unrefTimer is not a function
at Timeout.unref (https://deno.land/std@0.137.0/node/internal/timers.mjs:42:10)
at setUnrefTimeout (https://deno.land/std@0.137.0/node/timers.ts:32:36)
at Socket.setStreamTimeout [as setTimeout] (https://deno.land/std@0.137.0/node/internal/stream_base_commons.ts:339:22)
...
private output
...
at processTicksAndRejections (https://deno.land/std@0.137.0/node/_next_tick.ts:80:12)
```

**Comment by kt3k** [comment]
@jcs224 How do you try to use Bull MQ in Deploy? Is it via esm.sh? Do you import `https://esm.sh/bullmq`?

**Comment by jcs224** [comment]
I've been using the Node require polyfill, this works fine with the `--unstable` flag:

```ts
import { createRequire } from "https://deno.land/std@0.138.0/node/module.ts";
const require = createRequire(import.meta.url);

const { Queue } = require('bullmq')

const myQueue = new Queue('foo')
```

However, when I try this:
```ts
import { Queue } from 'https://esm.sh/bullmq'

const myQueue = new Queue('foo')

await myQueue.add('myJobName', { foo: 'bar' })
await myQueue.add('myJobName', { qux: 'baz' })
```

I get a weird lodash error
```
error: Uncaught SyntaxError: The requested module '/v78/lodash@4.17.21/deno/lodash.js' does not provide an export named 'get' at https://cdn.esm.sh/v78/bullmq@1.81.4/deno/bullmq.js:7:20803
```

Also, I've been doing this all with the CLI, not deploy. I know that Deploy only uses stable API so I didn't even bother to try. The node `require` polyfill should work in Deploy though, right? I should have probably checked that...

**Pull request merged**